Gems/ScriptCanvas/Code/Editor/Assets/ScriptCanvasFileHandling.cpp

```diff
@@ -49,6 +49,35 @@ namespace ScriptCanvasFileHandlingCpp
             }
         }
     }
+
+    class EntityIdMapper
+    {
+    public:
+        EntityIdMapper()
+        {
+            m_newIdsByOld[AZ::EntityId()] = AZ::EntityId();
+            m_newIdsByOld[ScriptCanvas::GraphOwnerId] = ScriptCanvas::GraphOwnerId;
+            m_newIdsByOld[ScriptCanvas::UniqueId] = ScriptCanvas::UniqueId;
+        }
+
+        AZ::EntityId GetNewId(const AZ::EntityId& old)
+        {
+            auto iter = m_newIdsByOld.find(old);
+            if (iter == m_newIdsByOld.end())
+            {
+                const AZ::EntityId newId = AZ::Entity::MakeId();
+                m_newIdsByOld.insert(AZStd::make_pair( old, newId ));
+                return newId;
+            }
+            else
+            {
+                return iter->second;
+            }
```

Comment:
Could simplify the logic since iter is only used in the else path:

```
            if (auto iter = m_newIdsByOld.find(old); iter != m_newIdsByOld.end())
            {
                return iter->second;
            }

            const AZ::EntityId newId = AZ::Entity::MakeId();
            m_newIdsByOld.insert(AZStd::make_pair( old, newId ));
            return newId;
```